packages/wrangler/src/dev.tsx

```diff
@@ -348,205 +354,69 @@ export async function startDev(args: StartDevOptions) {
-		let host = args.host || config.dev.host;
-		let zoneId: string | undefined;
-		const routes: Route[] | undefined =
-			args.routes || (config.route && [config.route]) || config.routes;
-
-		if (args.forceLocal) {
-			args.local = true;
-		}
-
-		if (!args.local) {
-			if (host) {
-				zoneId = await getZoneIdFromHost(host);
-			}
-			if (!zoneId && routes) {
-				const firstRoute = routes[0];
-				const zone = await getZoneForRoute(firstRoute);
-				if (zone) {
-					zoneId = zone.id;
-					host = zone.host;
-				}
-			}
-		} else if (!host) {
-			if (routes) {
-				const firstRoute = routes[0];
-				host = getHostFromRoute(firstRoute);
-			}
-		}
-
-		const nodeCompat = args.nodeCompat ?? config.node_compat;
-		if (nodeCompat) {
-			logger.warn(
-				"Enabling node.js compatibility mode for built-ins and globals. This is experimental and has serious tradeoffs. Please see https://github.com/ionic-team/rollup-plugin-node-polyfills/ for more details."
-			);
-		}
-
-		const getLocalPort = memoizeGetPort(DEFAULT_LOCAL_PORT);
-		const getInspectorPort = memoizeGetPort(DEFAULT_INSPECTOR_PORT);
-
-		const cliVars =
-			args.var?.reduce<Record<string, string>>((collectVars, v) => {
-				const [key, ...value] = v.split(":");
-				collectVars[key] = value.join("");
-				return collectVars;
-			}, {}) || {};
-
-		const cliDefines =
-			args.define?.reduce<Record<string, string>>((collectDefines, d) => {
-				const [key, ...value] = d.split(":");
-				collectDefines[key] = value.join("");
-				return collectDefines;
-			}, {}) || {};
-
 		// eslint-disable-next-line no-inner-declarations
 		async function getDevReactElement(configParam: Config) {
-			// now log all available bindings into the terminal
-			const bindings = await getBindings(configParam, {
-				kv: args.kv,
-				vars: { ...args.vars, ...cliVars },
-				durableObjects: args.durableObjects,
-				r2: args.r2,
-			});
-
-			// mask anything that was overridden in .dev.vars or cli args
-			// so that we don't log potential secrets into the terminal
-			const maskedVars = { ...bindings.vars };
-			for (const key of Object.keys(maskedVars)) {
-				if (maskedVars[key] !== configParam.vars[key]) {
-					// This means it was overridden in .dev.vars or cli args
-					// so let's mask it
-					maskedVars[key] = "(hidden)";
-				}
-			}
-
-			printBindings({
-				...bindings,
-				vars: maskedVars,
-			});
-
-			const assetPaths =
-				args.assets || config.assets
-					? getAssetPaths(config, args.assets)
-					: getSiteAssetPaths(
-							config,
-							args.site,
-							args.siteInclude,
-							args.siteExclude
-					  );
+			const { assetPaths, bindings } = await getBindingsAndAssetPaths(
+				args,
+				configParam
+			);
 
 			return (
 				<Dev
-					name={getScriptName({ name: args.name, env: args.env }, config)}
-					noBundle={!(args.bundle ?? !config.no_bundle)}
+					name={getScriptName({ name: args.name, env: args.env }, configParam)}
```

Comment:
@threepointone - I noticed we were using `config` here even though we're passing `configParam` into the component, hence the change

Reply:
huh I wonder what the behaviour is if we change the name during dev. would be nice if this works.

Reply:
good catch. this might have broken live reload on wrangler.toml silently, maybe you fixed it